app/domain/conjur/fetch_trusted_proxies.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require 'command_class'
+
+module Conjur
+
+  FetchTrustedProxies ||= CommandClass.new(
+    dependencies: {
+        role_cls:                 ::Role,
+        validate_account_exists:  ::Authentication::Security::ValidateAccountExists.new,
+        logger:                   Rails.logger
+    },
+    inputs: %i(account)
+  ) do
+
+    def call
+      validate_account_exists
+      fetch_trusted_proxies_list
+      delete_duplications_in_list
+      trusted_proxies_list
+    rescue => e
+      raise Errors::Conjur::TrustedProxiesMissing.new(
```

Comment:
It seems this error will be raised also for DB error (or any other error).
Is it really trusted proxies missing?
Also - why is it an error to not have trusted proxies? it is a valid configuration AFAIK

Reply:
Lets talk about this with @shaharglazner and @InbalZilberman as far as i understand this security feature is always enabled so if you are unable to fetch the trusted proxies from the DB we should fail the request

Reply:
I think we should distinguish here between the case of DB failure, and the case of unconfigured trusted proxies
DB failure case should be handled as an exception.
Unconfigured trusted proxies handling should be defined by Inbal and Shahar

Reply:
I do not think we should distinguish between the errors in this flows, any errors that prevent from me to return the trusted proxy list should be wrapped under `TrustedProxiesFetchFailed` error

Reply:
this is a new behaviour in our code. In other places we do not put an error inside an error in case it's related. The `account` comes from the authentication request so if I send a request with a wrong account i want the authentication error in the log to be concise and point me to the error. what do i care that you found that the account is missing while you are fetching the proxies and not, for example, while you are verifying that the webservice exists? 

For me as a user, all i want to know is that the account is incorrect and that i need to fix it.

Reply:
the stack trace will state the place where the error was raised but the one-liner should only say that the account doesn't exist IMO

Reply:
I agree that this is a new behaviour in our code , and that we decide about distinguishing between 2 cases. I think that we find that account is missing during fetching the proxies while verification